core/src/emuwidget.cpp

```diff
@@ -0,0 +1,187 @@
+	: QWidget(parent)
+	, m_ui(new Ui::EmuWidget)
+	, m_emuPath(path)
+	, m_enableDemo(false)
+	, m_emuProcess(nullptr)
+{
+	m_ui->setupUi(this);
+	m_emuProcess = new QProcess(this);
+	connect(m_ui->comboBoxDemoOption, &QComboBox::currentTextChanged, this, &EmuWidget::onOptionChanged);
+	init();
+
+	connect(m_ui->btnEnableDemo, &QPushButton::clicked, this, &EmuWidget::onEnableDemoClicked);
+	connect(m_ui->btnXmlPathBrowse, &QPushButton::clicked, this, [=](){
+		browseFile(m_ui->lineEditXmlPath);
+	});
+	connect(m_ui->btnRxTxDeviceBrowse, &QPushButton::clicked, this, [=](){
+		browseFile(m_ui->lineEditRxTxDevice);
+	});
+}
+
+EmuWidget::~EmuWidget()
+{
+	delete m_ui;
+	if (m_emuProcess) {
+		m_emuProcess->kill();
+	}
+}
+
+void EmuWidget::init()
+{
+	m_emuPath = findEmuPath();
+	m_emuPath.isEmpty() ? setStatusMessage("Can't find iio-emu in the system!")
+			    : setStatusMessage("");
+	this->setEnabled(!m_emuPath.isEmpty());
+
+	QMovie *loadingIcon(new QMovie(this));
+	loadingIcon->setFileName(":/gui/loading.gif");
+	m_ui->btnEnableDemo->setAnimation(loadingIcon);
+
+	m_ui->btnEnableDemo->setProperty("blue_button", QVariant(true));
+	m_ui->btnXmlPathBrowse->setProperty("blue_button", QVariant(true));
+	m_ui->btnRxTxDeviceBrowse->setProperty("blue_button", QVariant(true));
+	m_ui->btnEnableDemo->setAutoDefault(true);
+	m_ui->lineEditUri->setPlaceholderText("ip:127.0.0.1");
+	for (const QString &item : m_availableOptions) {
+		m_ui->comboBoxDemoOption->addItem(item);
+	}
+}
+
+QString EmuWidget::getEmuDeviceUri() const
+{
+	return m_ui->lineEditUri->text();
+}
+
+void EmuWidget::setStatusMessage(QString msg)
+{
+	m_ui->labelStatusMsg->clear();
+	m_ui->labelStatusMsg->setText(msg);
+}
+
+void EmuWidget::onEnableDemoClicked()
+{
+	m_ui->btnEnableDemo->startAnimation();
+	if (!m_enableDemo) {
+		bool started = startIioEmuProcess();
+		if (!started) {
+			stopEnableBtn("Enable Demo");
+			return;
+		}
+		stopEnableBtn("Disable Demo");
+		m_ui->widgetEnable->setEnabled(false);
+		if (m_ui->lineEditUri->text().isEmpty()) {
+			m_ui->lineEditUri->setText("ip:127.0.0.1");
+		}
+		m_enableDemo = !m_enableDemo;
+		Q_EMIT emuDeviceAvailable(m_ui->lineEditUri->text());
+	} else {
+		Q_EMIT disableEmuDevice();
+	}
+}
+
+QStringList EmuWidget::createArgList()
+{
+	QString option = m_ui->comboBoxDemoOption->currentText();
+	QStringList arguments;
+	arguments.append(option);
+	if (option.compare("generic") == 0) {
+		arguments.append(m_ui->lineEditXmlPath->text());
+		arguments.append(m_ui->lineEditRxTxDevice->text());
+	}
+	return arguments;
+}
+
+QString EmuWidget::findEmuPath()
+{
+	Preferences *p = Preferences::GetInstance();
+	QString pathFromPreferinces = p->get("iio_emu_path").toString();
+	pathFromPreferinces.replace(" ", "");
+	// iio-emu found in system
+	QString program = qgetenv("IIOEMU_BIN");
+	if (program.isEmpty()) {
+		// Env Var not found, try local folder
+		QString dirPath = QCoreApplication::applicationDirPath();
```

Comment:
this flow does not account for iio-emu being part of PATH - you just run it as

iio-emu